csrc/executor.cpp

```diff
@@ -2306,6 +2309,17 @@ void FusionExecutor::deserialize(
   // See table definition for FusionExecutor in serde/fusion_cache.fbs
 
   NVF_ERROR(buffer != nullptr, "serde::FusionExecutor is nullptr.");
+
+  // TODO Should we set fusion_id, concrete_id, runtime_id, and group_id when we
+  // skip compilation?
+  if (buffer->is_compilation_skipped()) {
+    fusion_ = std::make_unique<Fusion>(*fusion);
+    NVF_ERROR(
+        !isCompiled() && isCompilationSkipped(),
```

Comment:
`!isCompiled()` should be sufficient, why both the conditions?

Reply:
The assertion wasn't originally within the if-statement, and I didn't update it.